dev/release/README.md

```diff
@@ -17,7 +17,7 @@ specific language governing permissions and limitations
 under the License.
 -->
 
-# Comet Release Process
+# Comet Source Release Process
```

Comment:
What is Source means here?

Reply:
We are defining the process for making an official ASF release, which must be a source tarball.